### PR TITLE
drivers: dmic: update dmic flow initialization

### PR DESCRIPTION
**Pull request by dnikodem**
DMIC configuration should allow the use of periodic start.

Signed-off-by: Damian Nikodem <damian.nikodem@intel.com>

### CODE REVIEW

**Review by juimonen — DISMISSED** [review]
lgtm

**Review by singalsu — COMMENTED** [review]
Additions looks OK. Please check that the dmic_start/stop procedure with un-reset to start is compatible with this mode.

**Review by abonislawski — DISMISSED** [review]

File: drivers/dai/intel/dmic/dmic_nhlt.c [hunk 1]
```diff
@@ -344,11 +345,28 @@ int dai_dmic_set_config_nhlt(struct dai_intel_dmic *dmic, const void *bespoke_cf
 		return -EINVAL;
 	}
 
+	const uint32_t skip_regs[] = { (0x8 / 4), (0x10 / 4), (0x14 / 4), (0x18 / 4),
```
Comment (singalsu) [inline drivers/dai/intel/dmic/dmic_nhlt.c:348]: Why is this done?

Comment (dnikodem) [reply drivers/dai/intel/dmic/dmic_nhlt.c]: I removed this part - latest change set MIC_MUTE bit for unused PDM registers only. Thanks

**Review by jxstelter — APPROVED** [review]

**Review by abonislawski — APPROVED** [review]
It was tested in dmic part of our full scope tests?

**Comment by dnikodem** [comment]
> It was tested in dmic part of our full scope tests?

The previous version of this patch was tested in dmic range, I am waiting for the test results with the latest version. Thanks

**Review by marcinszkudlinski — APPROVED** [review]

**Comment by dnikodem** [comment]
> > It was tested in dmic part of our full scope tests?
> 
> The previous version of this patch was tested in dmic full scope, I am waiting for the test results with the latest version. Thanks

According to the latest test results with newest patch, I do not observe any new regression in dmic scope.

**Pull request merged**